config/options.go

```diff
@@ -445,12 +445,23 @@ func (o *Options) parseHeaders() error {
 			}
 
 		}
-		o.Headers = headers
-	} else if o.viperIsSet("headers") {
-		if err := o.viper.UnmarshalKey("headers", &headers); err != nil {
-			return fmt.Errorf("header %s failed to parse: %w", o.viper.Get("headers"), err)
+		o.SetResponseHeaders = headers
+		return nil
+	}
+
+	if o.viperIsSet("headers") {
+		log.Warn().Msg("config: headers has been renamed to set_response_headers and may be removed in the future")
+	}
+
+	// option was renamed from `headers` to `set_response_headers`. Both config settings are supported.
+	headerKeys := []string{"headers", "set_response_headers"}
+	for _, headerKey := range headerKeys {
+		if o.viperIsSet(headerKey) {
```

Comment:
I think we could also add a `header!=nil` check here to avoid looping again if values are found for a given key since we would overwrite the value anyway?

Reply:
Sorry I don't understand. What check do I need to add?

Reply:
I was just thinking it could be possible to break early if `headers` was already set vs the current behavior would check both, and then overwrite whatever was in `headers` with `set_response_headers` which probably what we want anyway.